### PR TITLE
Internationalisation

### PR DESCRIPTION
**Pull request by thepieterdc**
This PR adds some internationalisation features to the judge and to tests.

# Examples

### Get the language of the user in a test
```java
import dodona.i18n.Language;

@Test
public void test() {
    System.out.println("The current language is: " + Language.current());
}
```

### Provide an English and a Dutch version of a test description
**Tested method**
```java
@I18nTestDescription("my_tested_method")
public void test() {
    Assert.assertEquals("Hello", "World");
}
```

**evaluation/properties/descriptions.en.properties**
```properties
my_tested_method=My English Test
```

**evaluation/properties/descriptions.nl.properties**
```properties
my_tested_method=Mijn Nederlandse Test
```

## TODO
- Translate compilation errors/warnings
- Translate other dutch things

### CODE REVIEW

**Comment by thepieterdc** [comment]
@ninewise All that's left now is translate the compilation error/warning hints, along with other strings in the `run`-executable. Could you elaborate on how this can be implemented? I am unsure how to do this.

**Comment by ninewise** [comment]
For translating strings in sh; How about we add 2 files: `strings/nl.sh` and `strings/en.sh`. At the start of the `run` script, we `. "strings/${language}.sh`. The translation files could contain either printf strings in variables (e.g. `cannot_find_symbol='Je probeert een onbekende klasse "%s" te gebruiken. Mogelijks ben je de noodzakelijke import vergeten.'`) or functions (e.g. `cannot_find_symbol() { echo "Je probeert een onbekende klasse \"$1\" te gebruiken. Mogelijks ben je de noodzakelijke import vergeten."; }`) or something else still.

File: run [hunk 1]
```diff
@@ -246,10 +250,18 @@ if ! find "$resources" -name '*.java' | xargs javac -cp ".:${resources}:${workli
         'staff'
 fi
 
+# Find all .properties-files in the resources folder and copy them here
```
Comment (ninewise) [inline run]: Isn't it possible to just add the properties directory to the class path?

File: src/dodona/i18n/Language.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+     * passed.
+     *
+     * @return the current active language
+     */
+    public static Language current() {
+        final String fromProperties = System.getProperty(PROPERTY_LANGUAGE);
+        return Arrays.stream(Language.values())
+            .filter(lang -> Objects.equals(fromProperties, lang.identifier))
+            .findAny()
+            .orElse(ENGLISH);
```
Comment (ninewise) [inline src/dodona/i18n/Language.java]: I like.

File: src/dodona/junit/JSONListener.java [hunk 3]
```diff
@@ -1,28 +1,24 @@
...
-import dodona.feedback.StartTab;
-import dodona.feedback.StartContext;
-import dodona.feedback.StartTestcase;
-import dodona.feedback.CloseTab;
-import dodona.feedback.CloseContext;
-import dodona.feedback.CloseTestcase;
-import dodona.feedback.Message;
-import dodona.feedback.AppendMessage;
-import dodona.feedback.EscalateStatus;
+import dodona.feedback.*;
```
Comment (ninewise) [inline src/dodona/junit/JSONListener.java]: We don't use everything, so I prefer not to use a `*`.

File: src/dodona/junit/JSONListener.java [hunk 4]
```diff
@@ -1,28 +1,24 @@
...
+import dodona.i18n.I18nTestDescription;
+import dodona.i18n.Language;
 import dodona.json.Json;
-
```
Comment (ninewise) [inline src/dodona/junit/JSONListener.java]: Keep this newline to separate local, libs and stdlib.

File: src/dodona/junit/JSONListener.java [hunk 5]
```diff
@@ -106,6 +118,45 @@ public void aftertest(Failure failure) {
...
+     *
+     * @param desc the description
+     * @return the value of the I18nTestDescription annotation if available
+     */
+    private Optional<String> getI18nTestDescription(final Description desc) {
+        return Optional.ofNullable(this.descriptions).flatMap(bundle ->
+            Optional.ofNullable(desc.getAnnotation(I18nTestDescription.class))
+                .map(I18nTestDescription::value)
+                .filter(bundle::containsKey)
+                .map(bundle::getString)
```
Comment (ninewise) [inline src/dodona/junit/JSONListener.java]: Is this a Haskell-judge?

File: src/dodona/junit/JSONListener.java (hunk 3 above)
Comment (thepieterdc) [reply src/dodona/junit/JSONListener.java]: This was not intentional; my IDE performs import cleanup whenever I save a file. I have reversed this.

File: src/dodona/junit/JSONListener.java (hunk 4 above)
Comment (thepieterdc) [reply src/dodona/junit/JSONListener.java]: Same reason as above; my IDE cleans imports whenever I save files. I have reversed this change now.

File: run (hunk 1 above)
Comment (thepieterdc) [reply run]: I have tried to do so, but the Java judge does not seem to be able to find the file.

**Pull request merged**